internal/formatter/formatter_test.go

```diff
@@ -0,0 +1,132 @@
+package formatter
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-tableland/internal/tableland"
+)
+
+var rawJSON = []byte("{\"city\":\"dallas\"}")
+
+var input = &tableland.TableData{
+	Columns: []tableland.Column{
+		{Name: "name"},
+		{Name: "age"},
+		{Name: "location"},
+	},
+	Rows: [][]*tableland.ColumnValue{
+		{tableland.OtherColValue("bob"), tableland.OtherColValue(40), tableland.JSONColValue(rawJSON)},
+		{tableland.OtherColValue("jane"), tableland.OtherColValue(30), tableland.JSONColValue(rawJSON)},
+	},
+}
+
+var inputExtractable = &tableland.TableData{
+	Columns: []tableland.Column{
+		{Name: "name"},
+	},
+	Rows: [][]*tableland.ColumnValue{
+		{tableland.OtherColValue("bob")},
+		{tableland.OtherColValue("jane")},
+	},
+}
+
+var inputExtractable2 = &tableland.TableData{
+	Columns: []tableland.Column{
+		{Name: "location"},
+	},
+	Rows: [][]*tableland.ColumnValue{
+		{tableland.JSONColValue(rawJSON)},
+		{tableland.JSONColValue(rawJSON)},
+	},
+}
+
+func TestFormat(t *testing.T) {
+	type args struct {
+		userRows *tableland.TableData
+		output   Output
+		unwrap   bool
+		extract  bool
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "table",
+			args: args{userRows: input, output: Table},
+			want: "{\"columns\":[{\"name\":\"name\"},{\"name\":\"age\"},{\"name\":\"location\"}],\"rows\":[[\"bob\",40,{\"city\":\"dallas\"}],[\"jane\",30,{\"city\":\"dallas\"}]]}", //nolint
+		},
+		{
+			name: "objects",
+			args: args{userRows: input, output: Objects},
+			want: "[{\"name\":\"bob\",\"age\":40,\"location\":{\"city\":\"dallas\"}},{\"name\":\"jane\",\"age\":30,\"location\":{\"city\":\"dallas\"}}]", // nolint
+		},
+		{
+			name: "objects, extract",
+			args: args{userRows: inputExtractable, output: Objects, extract: true},
+			want: "[\"bob\",\"jane\"]",
+		},
+		{
+			name: "objects, extract nested json",
+			args: args{userRows: inputExtractable2, output: Objects, extract: true},
+			want: "[{\"city\":\"dallas\"},{\"city\":\"dallas\"}]",
+		},
+		{
+			name:    "objects, extract error",
+			args:    args{userRows: input, output: Objects, extract: true},
+			wantErr: true,
+		},
+		{
+			name: "objects, unwrap",
+			args: args{userRows: input, output: Objects, unwrap: true},
+			want: "{\"name\":\"bob\",\"age\":40,\"location\":{\"city\":\"dallas\"}}\n{\"name\":\"jane\",\"age\":30,\"location\":{\"city\":\"dallas\"}}\n", // nolint
+		},
+		{
+			name: "objects, extract, unwrap",
+			args: args{userRows: inputExtractable, output: Objects, extract: true, unwrap: true},
+			want: "\"bob\"\n\"jane\"",
+		},
+		{
+			name: "objects, extract, unwrap nested json",
+			args: args{userRows: inputExtractable2, output: Objects, extract: true, unwrap: true},
+			want: "{\"city\":\"dallas\"}\n{\"city\":\"dallas\"}",
+		},
```

Comment:
Covering lots of cases here, feeling confident that formatting works as expected.